spirv_msl.cpp

```diff
@@ -1231,16 +1231,20 @@ bool CompilerMSL::is_member_packable(SPIRType &ib_type, uint32_t index)
 
 	auto &mbr_type = get<SPIRType>(ib_type.member_types[index]);
 
-	// Only 3-element vectors or 3-row matrices need to be packed.
-	if (mbr_type.vecsize != 3)
+	// Only vectors or 3-row matrices need to be packed.
+	if (mbr_type.vecsize == 1 || (is_matrix(mbr_type) && mbr_type.vecsize != 3))
```

Comment:
What about matrices at weird offsets?

Reply:
As they are sort of arrays of vectors, packing rules might ban that. I guess relaxed_block_layout only relaxes alignment for non-array/matrix types?

Reply:
I checked the spec with and without the extension, and yeah, it only applies to vector types.